cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/monitor/RuntimeMonitor.java

```diff
@@ -90,22 +93,12 @@ protected void startUp() throws Exception {
   }
 
   private void initializeTopics() throws Exception {
-    Set<String> topicsToMonitor = new HashSet<>();
-    topicsToMonitor.addAll(Arrays.asList(cConf.get(Constants.RuntimeMonitor.TOPICS).split(",")));
-
-    HttpResponse response = restClient.execute(HttpMethod.POST, clientConfig.resolveURL("runtime/monitor/topics"),
-                                               GSON.toJson(topicsToMonitor), Collections.emptyMap(), null);
-
-    if (response.getResponseCode() == HttpURLConnection.HTTP_UNAVAILABLE) {
-      throw new ServiceUnavailableException(response.getResponseBodyAsString());
-    }
-
-    Map<String, String> configToTopic = GSON.fromJson(response.getResponseBodyAsString(StandardCharsets.UTF_8),
-                                                      MAP_STRING_STRING_TYPE);
+    Set<String> topicsConfigsToMonitor = new HashSet<>();
+    topicsConfigsToMonitor.addAll(Arrays.asList(cConf.get(Constants.RuntimeMonitor.TOPICS_CONFIGS).split(",")));
```

Comment:
The metrics could have multiple topics. You might want to refer to MessagingServiceUtils.getSystemTopics on how it is parsed (you could move that method to common if needed).

Reply:
Since we are using topicConfig per topic and keep track of offsets for each topic in RuntimeMonitor, this would require a new rest endpoint (because Runtime can be using different number of metrics than RuntimeMonitor). I will handle that in my next PR.

Reply:
ok. :)